### PR TITLE
organise bundles, add filetype checks and add Justin Cook's JS file

### PR DESCRIPTION
**Pull request by g-sam**
Is this OK by you @justin-cook-developer?

### CODE REVIEW

**Comment by justin-cook-developer** [comment]
Yeah, go for it!

**Comment by g-sam** [comment]
did I create a conflict with my own changes 😮 !

The emoji thing is unrelated to the file type thing.  you can enable emojis in the browser by uncommenting this line  https://github.com/dwiel/talon_community/pull/75/files#diff-db4e546e933d7ef4137433a92666a6dbR11.

For enabling, e.g. markdown editing in the browser you could add another entry to this tuple https://github.com/dwiel/talon_community/pull/75/files#diff-da30e1c1a6bc3076711ae9a0766ad580R9  to match something in the title. but then the function probably needs renaming.  alternatively,  you could remove the browsers from the list of context sensitive file types. that's what I thinking anyway, except that looking back over the code I think there's a mistake so that might not work at the moment.  it's a quick fix though if that's an acceptable temporary solution.

basically, these different ways of determining contexts need to be more composable than they are here, but this may be a start. I'm not sure if I will work out a proper solution to this soon as I need to get other bits of the scripts working for me and this will be my last night to work on it for a while.

**Comment by g-sam** [comment]
I have  corrected the error in is_filetype so that now you can remove a bundle from the file type sensitive bundle list in order to activate the context even if the file type does not match

**Comment by justin-cook-developer** [comment]
@dwiel what context would you like to add? I'm fine with adding other contexts. I only restricted it because I didn't want to mess with other things in the community repository and possibly cause naming conflicts between languages. Such as using the keyword 'state func'  for both a. python and javascript function

File: utils.py [hunk 1]
```diff
@@ -8,6 +8,9 @@
 import json
 import os
 
+from .bundle_groups import TERMINAL_BUNDLES, FILETYPE_SENSITIVE_BUNDLES
+
+VIM_IDENTIFIER = "(Vim)"
```
Comment (dwiel) [inline utils.py]: Duplicate line?

**Comment by dwiel** [comment]
Btw, meant we can merge after removing duplicate line. Was just musing about terminals. Thanks for awesome changes!

**Comment by g-sam** [comment]
fixed in https://github.com/dwiel/talon_community/pull/76 if that works for you?

**Comment by dwiel** [comment]
These contexts mostly make sense and I'm not sure what the best solution is. There are a lot of edge cases I find wanting these commands in "non-standard" places: talking about code in Slack, or in a browser, writing code inside a documentation file, writing code inside of a jupyter notebook, using markdown inside of a web-based to do app. I think the solution implemented here is it good stabbed at a reasonable middle-ground. Don't screw like you point out there may be some conflict between languages in contexts where the language isn't known.

**Comment by justin-cook-developer** [comment]
I understand exactly what you mean. I currently have my CSS file scope to both a browser and editor so i can use online editors. It seems like namespace and commands is the best thing to do, but is hard to find a shortening meaningful name space. I'm currently working on pulling out all of the technology specific commands from my javascript file in making my own directory which will be associated with my personal snippets. It will be a much more organized way to maintain technology specific snippets for vs code.

**Comment by justin-cook-developer** [comment]
* namespacing commands
* short and
